src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -426,10 +426,113 @@ public bool TryGetValue(ReadOnlySpan<byte> key, out long value, out EncodedKey e
                 value = default;
                 return false;
             }
+
             value = GetValue(ref state, state.LastSearchPosition);
             return true;
         }
 
+        public void InitializeStateForTryGetNextValue()
+        {
+            _internalCursor._pos = -1;
+            _internalCursor._len = 0;
+            PushPage(_state.RootPage, ref _internalCursor);
+            ref var state = ref _internalCursor._stk[_internalCursor._pos];
+            state.LastSearchPosition = 0;
+        }
+        
+        public bool TryGetNextValue(ReadOnlySpan<byte> key, out long value, out EncodedKey encodedKey)
+        {
+            ref var state = ref _internalCursor._stk[_internalCursor._pos];
+            if (state.Header->PageFlags == CompactPageFlags.Branch)
+            {
+                // the *previous* search didn't find a value, we are on a branch page that may
+                // be correct or not, try first to search *down*
+                encodedKey = EncodedKey.Get(key, this, state.Header->DictionaryId);
+                encodedKey = FindPageFor(ref _internalCursor, ref state, encodedKey);
+                state = ref _internalCursor._stk[_internalCursor._pos];
+
+                if (state.LastMatch == 0) // found it
+                    return ReturnValue(encodedKey, ref state, out value);
+                // did *not* find it, but we are somewhere on the tree that is ensured
+                // to be at the key location *or before it*, so we can now start scanning *up*
+            }
+            Debug.Assert(state.Header->PageFlags == CompactPageFlags.Leaf);
+            
+            encodedKey = EncodedKey.Get(key, this, state.Header->DictionaryId);
+
+            SearchInCurrentPage(encodedKey, ref state);
+            if (state.LastSearchPosition  >= 0) // found it, yeah!
+            {
+                value = GetValue(ref state, state.LastSearchPosition);
+                return true;
+            }
+
+            var pos = ~state.LastSearchPosition;
+            var shouldBeInCurrentPage = pos < state.Header->NumberOfEntries;
+
+            if (shouldBeInCurrentPage)
+            {
+                var nextEntry = GetEncodedKey(state.Page, state.EntriesOffsets[pos]);
+                var match = encodedKey.Encoded.SequenceCompareTo(nextEntry);
+                shouldBeInCurrentPage = match < 0;
+            }
+
+            if (shouldBeInCurrentPage == false)
```

Comment:
Minor - wouldn't be simpler to have `else` here instead?

Reply:
No, I see that `shouldBeInCurrentPage` is modified inside above if: `shouldBeInCurrentPage = match < 0;`